src/libudev/libudev-queue.c

```diff
@@ -106,17 +115,12 @@ _public_ struct udev_queue *udev_queue_unref(struct udev_queue *udev_queue)
  * udev_queue_get_udev:
  * @udev_queue: udev queue context
  *
- * Retrieve the udev library context the queue context was created with.
+ * This function is deprecated.
  *
- * Returns: the udev library context.
+ * Returns: NULL.
  **/
-_public_ struct udev *udev_queue_get_udev(struct udev_queue *udev_queue)
-{
-        if (udev_queue == NULL) {
-                errno = EINVAL;
-                return NULL;
-        }
-        return udev_queue->udev;
+_public_ struct udev *udev_queue_get_udev(struct udev_queue *udev_queue) {
+        return NULL;
```

Comment:
I didn't get all the way through the patch set, but the way I understand things, there is a major problem. The removal of the pointer to `struct udev` is a backwards incompatible change, and simply cannot be done without breaking backwards compatibility. Essentially, the symbol is being substituted with a stub instead of being removed. This *could* be OK if we were completely certain that there are not callers. But this is on old and very widely used library, so we can never be certain that there isn't some user, even in unpublished code. As a trivial example, somebody could have code like this:
```C
struct udev_queue* queue;

void init() {
     struct udev* = udev_new();
     /* do stuff with udev */
     queue = udev_queue_new(udev);
}
int main() {
       init();
       /* do stuff with queue */
       struct udev *udev = udev_queue_get_udev(queue);
       udev_free_queue(queue);
       udev_free(udev);
}
```

Similar reasoning applies to other patches which change the return value to NULL.